src/matrix.rs

```diff
@@ -47,6 +47,19 @@ impl<T: Num + PartialOrd + Copy> DynamicMatrix<T> {
         Self::fill(shape, T::one())
     }
 
+    pub fn reshape(&self, shape: &Shape) -> Result<DynamicMatrix<T>, ShapeError> {
+        if shape.order() != 2 {
```

Comment:
They would need to be three separate functions would they not? Because I can't have one method return either a `DynamicVector`, `DynamicMatrix` or `DynamicTensor` depending on the input shape. That's probably why this method is only to reshape into another matrix, the `flatten` method exists to convert it to a `DynamicVector`.

So would the problem be that we don't offer a method to convert to a Tensor?

Then you could do

```rust
matrix.to_tensor().reshape()
```

Reply:
After some discussion, we've decided that:
1. `reshape` will always return a `DynamicTensor`
2. `redimensions` will be used to return a `DynamicMatrix`

We think this is probably the most consistent way of doing things.